src/libraries/System.Private.Xml/tests/Xslt/XslCompiledTransformApi/System.Xml.Xsl.XslCompiledTransformApi.Tests.csproj

```diff
@@ -25,6 +32,5 @@
     <ProjectReference Include="$(CommonTestPath)System\Xml\ModuleCore\ModuleCore.csproj" />
     <ProjectReference Include="$(CommonTestPath)System\Xml\XmlCoreTest\XmlCoreTest.csproj" />
     <ProjectReference Include="$(CommonTestPath)System\Xml\XmlDiff\XmlDiff.csproj" />
-    <ProjectReference Include="..\XslTransformApi\System.Xml.Xsl.XslTransformApi.Tests.csproj" />
```

Comment:
why?

Reply:
@ViktorHofer because when we archive the `System.Xml.Xsl.XslTransformApi.Tests.csproj` we delete its output directory. Later on when this project is built it tries to reference the assembly but it's no longer there.

Reply:
Which means we shouldn't delete the outdir before all test projects are built. Deleting the appbundle makes sense but deleting the outdir breaks the up-to-date file checks when referencing such projects via P2Ps. There might be other cases in the future where we want to include a test project as part of another one.

Reply:
what would the use case for such a P2P reference be? I can only think of code sharing and we can do that on the source level (actually that's the main way we share test code right now).

Reply:
we can't delete after all test projects are build because we run out of disk space before that.

Reply:
Another option I was thinking, was having a property that test projects could set to opt-out from removing their outdir.

Reply:
Why would we want to delete the OutDir at all? It doesn't take up much space. Deleting the bundle I agree with as it's super large.

Reply:
The OutDir is part of the up-to-date check of msbuild. If we want to delete the OutDir at should do it via the Clean target which also deletes the intermediate folder and other file writes. But as said I don't see much value as the main issue is the bundle that isn't cleaned up.

Reply:
the AppBundle folder isn't really the largest part, the OutDir also includes the `publish` folder which contains the complete BCL. we can probably get under the the disk space limit by only removing the AppBundle folder if you feel strongly :)

Reply:
We could remove PublishDir and that will remove PublishDir and AppBundle.

Reply:
Yeah I believe removing only the bundle and the publish folder is the best path forward. I feel strongly about this as I would like to avoid yet another difference in CI vs local development as it caused numerous issues in the past and one recently with the partial artifacts state. I'm not concerned about the PublishDir or bundle as those aren't part of the msbuild freshness check.

Reply:
@ViktorHofer changed

Reply:
Thanks 👍 Can we also condition the BundleDir property on Browser so that it is skipped in the RemoveDir Task?

Reply:
Fixed